### PR TITLE
Bulk update rights from file of ids

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?
This allows a list of ids to be submitted, and to change the rights on each one to `no rights` or `chargeable`

## How to test
Run this on test against some images.

## How can we measure success?
easily. 

## Have we considered potential risks?
yes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +83,27 @@ EXAMPLES
...
+## `grid bulk:rights INPUT RIGHTS OUTPUT FAILURES`
+
+Reads a text file containing image ids, and sets their usage rights.
+
+```
+USAGE
+  $ grid bulk:rights INPUT RIGHTS OUTPUT FAILURES
+
+ARGUMENTS
+  INPUT     file to read
```
Comment (paperboyo) [inline README.md]: Worth specifying in Readme what is expected in the file?

**Comment by paperboyo** [comment]
As this may apply to potentially even hundreds of thousands of images, whose responsibility is it for it not to bring the world to its knees?

**Comment by paperboyo** [comment]
image

I liked this version of the answer better. Because it was correct answer. But a bit scary? Bulk API? Specifying the kinesis stream? ASCII-art progress-bar, maybe a part of this fancy lib?

**Comment by AWare** [comment]
It attempts them one at a time rather than in parallel, so shouldn't present too much of a burden.

**Comment by jennygrahamjones** [comment]
What is the best way to test new commands before they're incorporated into a release of the package?

**Comment by AWare** [comment]
I set my default grid to TEST and ran this against that.

File: src/commands/bulk/rights.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+  static rightsTuples = Object.entries(
+    BulkRights.rights
+  ).map(([name, data]) => [name, JSON.stringify(data)])
+
+  static flags = {
+    ...ApiCommand.flags,
+    help: flags.help({char: 'h'}),
+
+    // output: flags.string()
```
Comment (jennygrahamjones) [inline src/commands/bulk/rights.ts]: Is this needed?

File: README.md [hunk 3]
```diff
@@ -221,12 +263,14 @@ OPTIONS
 
   -d, --dryRun           Display the result of the reingestion only
 
+  -f, --force            Force reingestion if the image is already present in the Grid
+
```
Comment (jennygrahamjones) [inline README.md:271]: Were these all documentation updates that were missed in previous PRs?

**Review by jennygrahamjones — APPROVED** [review]
To clarify, I had to run `npm run local bulk:rights...`. Previously, this would have been `./bin/run bulk:rights...`. It's probably useful to add something about this to the README, for clarity on local development/testing.

That being said, the new command seems to work as described 👍

File: README.md (hunk 3 above)
Comment (AWare) [reply README.md:271]: Probably, I updated the readme using the script.

File: src/commands/bulk/rights.ts (hunk 2 above)
Comment (AWare) [reply src/commands/bulk/rights.ts]: No.

**Comment by AWare** [comment]
@jennygrahamjones how's my readme change?

**Comment by jennygrahamjones** [comment]
> @jennygrahamjones how's my readme change?

👍 👍 thanks!

**Pull request merged**